src/Microsoft.IdentityModel.Logging/LoggerContext.cs

```diff
@@ -28,9 +29,60 @@ public LoggerContext(Guid activityId)
             ActivityId = activityId;
         }
 
+#pragma warning disable CS3001 // ILogger is not CLSCompliant
+#pragma warning disable CS3003 // ILogger is not CLSCompliant
+        /// <summary>
+        /// Instantiates a new <see cref="LoggerContext"/>.
+        /// </summary>
+        /// <param name="logger"><see cref="ILogger"/> to record logs.</param>
+        /// <exception cref="ArgumentNullException">if <paramref name="logger"/> is null.</exception>
+        public LoggerContext(ILogger logger)
+        {
+            Logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        /// <summary>
+        /// Instantiates a new <see cref="LoggerContext"/> with an activity identifier.
+        /// </summary>
+        /// <param name="logger"><see cref="ILogger"/> to record logs.</param>
+        /// <param name="activityId">activity id to include in logs.</param>
+        /// <exception cref="ArgumentNullException">if <paramref name="logger"/> is null.</exception>
+        public LoggerContext(ILogger logger, Guid activityId)
+        {
+            Logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            ActivityId = activityId;
+        }
+
+        /// <summary>
+        /// Instantiates a new <see cref="LoggerContext"/> with a correlation identifier.
+        /// </summary>
+        /// <param name="logger"><see cref="ILogger"/> to record logs.</param>
+        /// <param name="correlationId">correlation id to include in logs.</param>
+        /// <exception cref="ArgumentNullException">if <paramref name="logger"/> is null.</exception>
+        public LoggerContext(ILogger logger, string correlationId)
+        {
+            Logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            CorrelationId = correlationId;
```

Comment:
We could check if the correlation id is a guid, if it is, then set the activity id with it